### PR TITLE
Remove API calls from ThreadController

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #459

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 1]
```diff
@@ -302,14 +300,76 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
         refreshThreads()
     }
 
+    //region Open Thread
+    fun openThread(threadUid: String) = liveData(Dispatchers.IO) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:304]: Why did you move it to here?

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:306]: Because I moved `markAsSeen()` from ThreadController to MainViewModel.

**Review by sirambd — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]
